Modules/_sqlite/statement.c

```diff
@@ -139,23 +131,25 @@ stmt_traverse(pysqlite_Statement *self, visitproc visit, void *arg)
 }
 
 /*
- * Checks if there is anything left in an SQL string after SQLite compiled it.
- * This is used to check if somebody tried to execute more than one SQL command
- * with one execute()/executemany() command, which the DB-API and we don't
- * allow.
+ * Strip leading whitespace and comments from SQL string and return a
+ * pointer to the first non-whitespace, non-comment character.
+ *
+ * This is used to check if somebody tries to execute more than one SQL query
+ * with one execute()/executemany() command, which the DB-API don't allow.
  *
- * Returns 1 if there is more left than should be. 0 if ok.
+ * It is also used to harden DML query detection.
  */
-static int pysqlite_check_remaining_sql(const char* tail)
+static inline const char *
+lstrip_sql(const char *sql)
 {
-    const char* pos = tail;
+    const char *pos = sql;
 
     parse_remaining_sql_state state = NORMAL;
 
     for (;;) {
         switch (*pos) {
             case 0:
-                return 0;
+                return NULL;
```

Comment:
Note to self: this makes the `return NULL` at the end of the function unreachable. This can be easily fixed with a tiny refactor:

```c
lstrip_sql(const char *sql)
{
    parse_remaining_sql_state state = NORMAL;

    for (const char *pos; *pos; pos++) {
        ...
    }

    return NULL;
```

I'll add that in a separate PR.